HTTP Client Plugin example does not compile

**Describe the bug**
The HTTP Client Plugin sample provided at https://godoc.org/github.com/devopsfaith/krakend/transport/http/client/plugin does not compile.

**To Reproduce**
Steps to reproduce the behavior:
1. go version go1.11.13 linux/amd64
2. Copy source from https://godoc.org/github.com/devopsfaith/krakend/transport/http/client/plugin
3. Build with `go build -buildmode=plugin -o client-plugin.so`
4. Get error: `./main.go:33:9: cannot use func literal (type func(http.ResponseWriter, *http.Request)) as type http.Handler in return argument: func(http.ResponseWriter, *http.Request) does not 
implement http.Handler (missing ServeHTTP method)`

**Expected behavior**
It should compile

**Logs**
NA

**Additional context**
The issue can be fixed changing the return func line to wrap the returned function in `http.HandlerFunc`:

```go
return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
		fmt.Fprintf(w, "Hello, %q", html.EscapeString(req.URL.Path))
	}), nil
```

Maybe this is related to go version? I understand krakend requires go 1.11 (https://github.com/devopsfaith/krakend-ce#build-requirements) so would expect this to compile.

--- a/transport/http/client/plugin/doc.go
+++ b/transport/http/client/plugin/doc.go
@@ -34,9 +34,9 @@
 // 			return nil, fmt.Errorf("unknown register %s", name)
 // 		}
 //		// return the actual handler wrapping or your custom logic so it can be used as a replacement for the default http client
-// 		return func(w http.ResponseWriter, req *http.Request) {
+// 		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 // 			fmt.Fprintf(w, "Hello, %q", html.EscapeString(req.URL.Path))
-// 		}, nil
+// 		}), nil
 // 	}
 //
 // 	func init() {

